### PR TITLE
Make CTFrame draw at path origin

### LINKED ISSUE #2571: CTFramesetterCreate Ignores CGPath Parameter
It appears that the CTFramesetterCreate API doesn't take into consideration the location of the CGPath parameter passed in.

### PR DESCRIPTION
**Pull request by aballway**
Fixes #2571

### CODE REVIEW

File: tests/unittests/CoreGraphics.drawing/ImageComparison.h [hunk 1]
```diff
@@ -34,7 +35,7 @@ struct Pixel {
         return !(*this == o);
     }
     bool near(const Pixel& o) const {
-        return (abs(r - o.r) + abs(g - o.g) + abs(b - o.b) + abs(a - o.a)) < sc_visualThreshold;
+        return (std::abs(r - o.r) + std::abs(g - o.g) + std::abs(b - o.b) + std::abs(a - o.a)) < sc_visualThreshold;
```
Comment (aballway) [inline tests/unittests/CoreGraphics.drawing/ImageComparison.h]: Also fixes this, which was not building on reference platform

Comment (DHowett-MSFT) [inline tests/unittests/CoreGraphics.drawing/ImageComparison.h]: y

File: tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp [hunk 2]
```diff
@@ -684,3 +684,29 @@ TEXT_DRAW_TEST_P(RTLText, DrawRTLText) {
...
+    CGContextSetTextMatrix(context, CGAffineTransformRotate(textMatrix, 45 * M_PI / 180.0));
+    CGContextScaleCTM(context, 1.1, 1.3);
+    // Create style setting to match given alignment
+    CTParagraphStyleSetting setting[1];
+    CTTextAlignment alignment = kCTLeftTextAlignment;
+    setting[0].spec = kCTParagraphStyleSpecifierAlignment;
+    setting[0].valueSize = sizeof(CTTextAlignment);
+    setting[0].value = &alignment;
+
+    woc::unique_cf<CTParagraphStyleRef> paragraphStyle{ CTParagraphStyleCreate(setting, std::extent<decltype(setting)>::value) };
```
Comment (DHowett-MSFT) [inline tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp]: nit: switch these to StrongCF with TakeOwnership

File: Frameworks/CoreText/CTFrame.mm [hunk 3]
```diff
@@ -126,7 +126,9 @@ void CTFrameDraw(CTFrameRef frame, CGContextRef ctx) {
 
         // Invert Text Matrix and CTM so glyphs are drawn in correct orientation and position
         CGAffineTransform textMatrix = CGContextGetTextMatrix(ctx);
-        CGContextSetTextMatrix(ctx, CGAffineTransformMake(textMatrix.a, -textMatrix.b, textMatrix.c, -textMatrix.d, 0, 0));
+        CGContextSetTextMatrix(
+            ctx,
+            CGAffineTransformMake(textMatrix.a, -textMatrix.b, textMatrix.c, -textMatrix.d, boundingRect.origin.x, -boundingRect.origin.y));
```
Comment (msft-Jeyaram) [inline Frameworks/CoreText/CTFrame.mm]: add a comment on what's going on

File: tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp [hunk 4]
```diff
@@ -684,3 +684,29 @@ TEXT_DRAW_TEST_P(RTLText, DrawRTLText) {
 }
 
 INSTANTIATE_TEST_CASE_P(RTLTextDrawing, RTLText, ::testing::ValuesIn(c_rotations));
+
+TEXT_DRAW_TEST_F(CTDrawing, CGPathTranslation, WhiteBackgroundTest<>) {
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    // Creates path with current rectangle
+    woc::unique_cf<CGMutablePathRef> path{ CGPathCreateMutable() };
```
Comment (msft-Jeyaram) [inline tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp]: >CGMutablePathRef [](start = 19, length = 16)

nit: Use StrongCF

File: tests/unittests/CoreGraphics.drawing/ImageComparison.h (hunk 1 above)
Comment (aballway) [inline tests/unittests/CoreGraphics.drawing/ImageComparison.h]: It was confused about which abs for some reason

File: tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp [hunk 5]
```diff
@@ -684,3 +684,29 @@ TEXT_DRAW_TEST_P(RTLText, DrawRTLText) {
...
+    CGPathAddRect(path.get(), nullptr, { { 25, 50 }, bounds.size });
+
+    CGAffineTransform textMatrix = CGContextGetTextMatrix(context);
+    CGContextSetTextMatrix(context, CGAffineTransformRotate(textMatrix, 45 * M_PI / 180.0));
+    CGContextScaleCTM(context, 1.1, 1.3);
```
Comment (MSFTFox) [inline tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp]: These transformations seem a bit unnecessary to test the path location in isolation. I would be a bit more confident in the test without them, or one without them in addition to this showing it works in isolation and in conjunction with transforms.

Comment (aballway) [inline tests/unittests/CoreGraphics.drawing/CTDrawingTests.cpp]: I wanted to guarantee that the path's origin would be properly affected by the transforms, and I don't think a basic test would give us any more coverage.

**Pull request merged**